llvm/lib/Analysis/AssumeBundleQueries.cpp

```diff
@@ -180,6 +180,10 @@ llvm::getKnowledgeForValue(const Value *V,
     }
     return RetainedKnowledge::none();
   }
+
+  if (!V->hasUseList())
+    return RetainedKnowledge::none();
```

Comment:
As a followup, we should completely remove the non-AC code in this function. We shouldn't be querying assumes without AC.